script.rb

```diff
@@ -0,0 +1,93 @@
+module Enumerable
+  def my_each
+    return to_enum(:each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr.length.times { |item| yield(arr[item]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:each_with_index) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr.length.times { |item| yield(arr[item], item) }
+    self
+  end
+
+  def my_select
+    new_arr = []
+    return to_enum(:select) unless block_given?
+
+    my_each { |m| new_arr << m if yield(m) }
+    new_arr
+  end
```

Comment:
#my_select returns an array containing all elements of enum for which the given block returns 
a true value::range